src/Turbo/src/Helper/TurboStream.php

```diff
@@ -86,12 +86,34 @@ public static function refresh(?string $requestId = null): string
         return \sprintf('<turbo-stream action="refresh" request-id="%s"></turbo-stream>', htmlspecialchars($requestId));
     }
 
-    private static function wrap(string $action, string $target, string $html, string $attr = ''): string
+    /**
+     * Custom action and attributes.
+     *
+     * @param array<string, string|int|null> $attr
+     */
+    public static function custom(string $action, string $target, string $html, array $attr = []): string
     {
+        if (\array_key_exists('action', $attr) || \array_key_exists('targets', $attr)) {
+            throw new \InvalidArgumentException('The "action" and "targets" attributes are reserved and cannot be used.');
+        }
+
+        $attr['action'] = $action;
+        $attr['targets'] = $target;
+        $attr = array_merge(['action' => $action, 'targets' => $target], $attr);
+
+        $attrString = '';
+        foreach ($attr as $key => $value) {
+            if (null === $value) {
+                $attrString .= \sprintf(' %s', $key);
+            } else {
+                $attrString .= \sprintf(' %s="%s"', $key, \is_string($value) ? htmlspecialchars($value) : $value);
```

Comment:
If $value is a Stringable object, this can be dangerous.

Special cases for boolean should be added too probably.

Reply:
I've changed it so it covers the types specified in the typehint and would fail if an object is passed